### PR TITLE
feat: projections and sqlite

### PR DESCRIPTION
**Pull request by lucaelin**

### CODE REVIEW

**Comment by lucaelin** [comment]
todo: 
restore journal event preprocessing (like autoscan) into PromptGenerator 
restore timestamp clock offset to in-game time inside of PromptGenerator 
restore ship status from journal into a projection
add delta event projection updates for missions and similar event types
read through remaining todo comments and fix those

File: src/Chat.py [hunk 1]
```diff
@@ -86,46 +88,9 @@ def reply(client, events: List[Event], new_events: List[Event], prompt_generator
...
-            eventManager.add_game_event(item['event_content'])
-            current_status['extra_events'].pop(0)
-
-    # Update previous status
-    previous_status = current_status
-
-
-jn = None
-tts = None
+backstory: str = None
```
Comment (RatherRude) [inline src/Chat.py:91]: not global?

File: src/lib/EDJournal.py [hunk 2]
```diff
@@ -1,70 +1,97 @@
...
+    def _reading_thread(self):
+        backoff = 1
+        while True:
+            try: 
+                self._reading_loop()
+            except Exception as e:
+                log('error', 'An error occurred during journal processing', e, traceback.format_exc())
+                sleep(backoff)
+                log('info', 'Attempting to restart journal processing after failure')
+                backoff *= 2
```
Comment (RatherRude) [inline src/lib/EDJournal.py:62]: who resets this?

File: src/lib/STT.py [hunk 3]
```diff
@@ -89,7 +92,7 @@ def _listen_continuous_thread(self):
             try: 
                 self._listen_continuous_loop()
             except Exception as e:
-                log('error', 'An error occurred during speech recognition', e)
+                log('error', 'An error occurred during speech recognition', e, traceback.format_exc())
                 sleep(backoff)
                 log('info', 'Attempting to restart speech recognition after failure')
                 backoff *= 2
```
Comment (RatherRude) [inline src/lib/STT.py:98]: who resets this?

File: src/lib/TTS.py [hunk 4]
```diff
@@ -38,7 +40,7 @@ def _playback_thread(self):
             try: 
                 self._playback_loop()
             except Exception as e:
-                log('error', 'An error occurred during speech synthesis', e)
+                log('error', 'An error occurred during speech synthesis', e, traceback.format_exc())
                 sleep(backoff)
                 log('info', 'Attempting to restart audio playback after failure')
                 backoff *= 2
```
Comment (RatherRude) [inline src/lib/TTS.py:46]: who resets this?

Comment (RatherRude) [inline src/lib/TTS.py:46]: who resets this?

File: src/lib/STT.py (hunk 3 above)
Comment (lucaelin) [reply src/lib/STT.py:98]: not related to this MR

File: src/lib/TTS.py (hunk 4 above)
Comment (lucaelin) [reply src/lib/TTS.py:46]: not related to this MR

Comment (lucaelin) [reply src/lib/TTS.py:46]: not related to this MR

File: src/lib/EDJournal.py (hunk 2 above)
Comment (lucaelin) [reply src/lib/EDJournal.py:62]: analog implementation to other places

**Pull request merged**